### PR TITLE
Add initial location based auth methods

### PR DESCRIPTION
**Pull request by jkeck**
Connects to #8 

I believe the current behavior would account for things requiring Stanford **OR** Location, but would not restrict location based content to **ONLY** Stanford users. (See #9)

### CODE REVIEW

File: lib/dor/rights_auth.rb [hunk 1]
```diff
@@ -115,6 +115,22 @@ def stanford_only_rights
       [@obj_lvl.group[:stanford].value, @obj_lvl.group[:stanford].rule]
     end
 
+    def location_rights(location_name)
```
Comment (ndushay) [inline lib/dor/rights_auth.rb]: should this have comments like the analogous methods do?

File: lib/dor/rights_auth.rb [hunk 2]
```diff
@@ -152,6 +168,15 @@ def stanford_only_rights_for_file(file_name)
       [@file[file_name].group[:stanford].value, @file[file_name].group[:stanford].rule]
     end
 
+    def location_rights_for_file(file_name, location_name)
```
Comment (ndushay) [inline lib/dor/rights_auth.rb]: should this method have comments?

**Comment by ndushay** [comment]
presumably tests would fail if this PR wasn't backwards compatible?

**Comment by ndushay** [comment]
also, a coupla line comments here ...

**Comment by jmartin-sul** [comment]
@jkeck: could you give an explanation about what `@obj_lvl` and `obj_lvl` are?  it's not immediately obvious, having not looked at the dor-rights-auth code much in the past.

**Comment by tingulfsen** [comment]
Admittedly I did some pairing early on this, but I am 👍

**Comment by jkeck** [comment]
@jmartin-sul if I follow what you're asking:

`obj_lvl` is used in a class method (`Dor::Rights::Auth.parse`) **on** an instance of the class to set the `@obj_lvl` instance variable used in the instance methods (`agent_rights`, `location_rights`, etc).

What is stored in the `@obj_lvl` variable are object level rules (i.e. the rules that are not qualified by an `<access>` block that includes a `<file>` element).

**Pull request merged**